src/radical/pilot/agent/executing/parsl_rp.py

```diff
@@ -351,16 +352,9 @@ def submit(self, func, *args, **kwargs):
         Kwargs:
             - **kwargs (dict) : A dictionary of arbitrary keyword args for func.
         """
-        self._task_counter += 1
-        # FIXME: use uid generator
-        task_id = 'task.parsl.%06d' % self._task_counter
-
-      # # ----------------------------------------------------------------------
-      # # test code: this is the fastest possible executor implementation
-      # self.future_tasks[task_id] = Future()
-      # self.future_tasks[task_id].set_result(3)
-      # return self.future_tasks[task_id]
-      # # ----------------------------------------------------------------------
+        self._task_counter = ru.generate_id('task.%(item_counter)06d',
```

Comment:
Why do we need the `self._task_counter variable?

Reply:
For two reasons: 

1. Maintain consistency in terms of task numbers between RP and Parsl (we create a future that the DFK needs to look for when fetching the task info if it's finished/pending/running).
2. I followed the class code template provided by `HTEX` for `RP-Parsl` executor which used `self._task_counter`.

Reply:
Ohh, I got it and makes sense. I addressed your proposal in the latest commit.